sqlglot/dialects/dialect.py

```diff
@@ -42,6 +42,20 @@ class Dialects(str, Enum):
 class _Dialect(type):
     classes: t.Dict[str, t.Type[Dialect]] = {}
 
+    def __eq__(cls, other: DialectType) -> bool:  # type: ignore
+        if cls is other:
+            return True
+        if isinstance(other, str):
+            return cls is cls.get(other.lower())
```

Comment:
i don't think you need to call lower, no one should be calling SNOWFLAKE

Reply:
fixed